### PR TITLE
Update API docs

### PR DESCRIPTION
**Pull request by flamber**
- Was a little outdated, and I don't think there's going to be more API changes before 0.35.0
- Removed the version number, since people are spending time trying to find one for their version
- Added note that it's generated from source, so people don't try to update the docs manually and I don't have to search for how to generate it again.

### CODE REVIEW

File: docs/api-documentation.md [hunk 1]
```diff
@@ -849,34 +847,51 @@ Delete a `Database`.
...
+  * `include=tables` means we should hydrate the Tables belonging to each DB. Default: `false`.
+
+  * `saved` means we should include the saved questions virtual database. Default: `false`.
+
+  * `include_tables` is a legacy alias for `include=tables`, but should be considered deprecated as of 0.35.0, and will
+    be removed in a future release.
+
+  * `include_cards` here means we should also include virtual Table entries for saved Questions, e.g. so we can easily
+    use them as source Tables in queries. This is a deprecated alias for `saved=true` + `include=tables` (for the saved
+    questions virtual DB). Prefer using `include` and `saved` instead.
```
Comment (mazameli) [inline docs/api-documentation.md]: Looks like there's some funky unnecessary whitespace in lines 856–861.

File: docs/api-documentation.md [hunk 2]
```diff
@@ -849,34 +847,51 @@ Delete a `Database`.
...
+    use them as source Tables in queries. This is a deprecated alias for `saved=true` + `include=tables` (for the saved
+    questions virtual DB). Prefer using `include` and `saved` instead. 
 
 ##### PARAMS:
 
 *  **`include_tables`** value may be nil, or if non-nil, value must be a valid boolean string ('true' or 'false').
 
 *  **`include_cards`** value may be nil, or if non-nil, value must be a valid boolean string ('true' or 'false').
 
+*  **`include`** null
```
Comment (mazameli) [inline docs/api-documentation.md]: Is this all this needs to say?

File: docs/api-documentation.md [hunk 3]
```diff
@@ -849,34 +847,51 @@ Delete a `Database`.
...
+
+*  **`saved`** value may be nil, or if non-nil, value must be a valid boolean string ('true' or 'false').
+
 
 ## `GET /api/database/:id`
 
-Get `Database` with ID.
+Get a single Database with `id`. Optionally pass `?include=tables` or `?include=tables.fields` to include the Tables
+  belonging to this database, or the Tables and Fields, respectively.
```
Comment (mazameli) [inline docs/api-documentation.md]: Seeing unnecessary newline and spaces here.

File: docs/api-documentation.md [hunk 4]
```diff
@@ -849,34 +847,51 @@ Delete a `Database`.
...
 
 ## `GET /api/database/:id/autocomplete_suggestions`
 
-Return a list of autocomplete suggestions for a given PREFIX.
-   This is intened for use with the ACE Editor when the User is typing raw SQL.
-   Suggestions include matching `Tables` and `Fields` in this `Database`.
+Return a list of autocomplete suggestions for a given `prefix`.
+
+  This is intened for use with the ACE Editor when the User is typing raw SQL. Suggestions include matching `Tables`
+  and `Fields` in this `Database`.
```
Comment (mazameli) [inline docs/api-documentation.md]: Unnecessary newline and spaces.

File: docs/api-documentation.md [hunk 5]
```diff
@@ -1468,12 +1507,16 @@ Update `Field` with ID.
 ## `GET /api/geojson/:key`
 
 Fetch a custom GeoJSON file as defined in the `custom-geojson` setting. (This just acts as a simple proxy for the
-  file specified for KEY).
+  file specified for `key`).
```
Comment (mazameli) [inline docs/api-documentation.md]: Unnecessary newline and spaces.

File: docs/api-documentation.md [hunk 6]
```diff
@@ -1950,6 +1989,15 @@ Delete a Pulse. (DEPRECATED -- don't delete a Pulse anymore -- archive it instea
 *  **`id`** 
 
 
+## `DELETE /api/pulse/:id/subscription/email`
+
+For uses to remove themselves from a pulse subscription
```
Comment (mazameli) [inline docs/api-documentation.md]:
```suggestion
For users to unsubscribe themselves from a pulse or alert subscription.
```

File: docs/api-documentation.md [hunk 7]
```diff
@@ -2319,8 +2363,8 @@ You must be a superuser to do this.
 
 ## `POST /api/setup/`
 
-Special endpoint for creating the first user during setup.
-   This endpoint both creates the user AND logs them in and returns a session ID.
+Special endpoint for creating the first user during setup. This endpoint both creates the user AND logs them in and
+  returns a session ID.
```
Comment (mazameli) [inline docs/api-documentation.md]: Unnecessary newline and spaces.

**Review by mazameli — CHANGES_REQUESTED** [review]

**Comment by flamber** [comment]
@mazameli All those funky newlines and spaces comes from the source code. So would we need to rewrite all the source comments, so it would generate a more sane docs? Example:
https://github.com/metabase/metabase/blob/3d5bd0a59a6514c1a6f4a22496e42534860f807e/src/metabase/api/setup.clj#L40-L41

**Comment by mazameli** [comment]
> All those funky newlines and spaces comes from the source code. So would we need to rewrite all the source comments, so it would generate a more sane docs?

Gotcha. I should have checked this initially, but it looks like the whitespace is ignored when the docs are rendered, so I guess leaving the comments as-is is okay.

File: docs/api-documentation.md (hunk 2 above)
Comment (flamber) [reply docs/api-documentation.md]: No, I think there might be something wrong with the generator:
https://github.com/metabase/metabase/blob/e1746b4f66f6dc77a4acca84be3f403b0cd48853/src/metabase/api/database.clj#L190
I guess it should generate something like this?
``` *  **`include`** value may be nil, or if non-nil, value must be `tables` ```
Fixing it manually will remove it next time we run it, so we need to fix the generator.
But I don't understand how the generator works - Clojure.

Comment (flamber) [reply docs/api-documentation.md]:
```suggestion
*  **`include`** value may be nil, or if non-nil, value must be `tables`
```

**Review by dacort — APPROVED** [review]
